wolfCLU/src/certgen/clu_certgen_ecc.c

```diff
@@ -1,53 +1,82 @@
-#include "clu_include/certgen/clu_certgen.h"
+/* clu_certgen_ecc.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL. (formerly known as CyaSSL)
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+#include <wolfclu/clu_header_main.h>
+#include <wolfclu/certgen/clu_certgen.h>
 
 void free_things_ecc(byte** a, byte** b, byte** c, ecc_key* d, ecc_key* e,
                                                                      WC_RNG* f);
 
 int make_self_signed_ecc_certificate(char* keyPath, char* certOut, int oid) {
     int ret = 0;
+    int certBufSz;
     word32 index = 0;
-    
+
     Cert newCert;
+    char country[3];
+    char province[CTC_NAME_SIZE];
+    char city[CTC_NAME_SIZE];
+    char org[CTC_NAME_SIZE];
+    char unit[CTC_NAME_SIZE];
+    char commonName[CTC_NAME_SIZE];
+    char email[CTC_NAME_SIZE];
+    char daysValid[CTC_NAME_SIZE];
+
     ecc_key key;
     WC_RNG rng;
-    
+
     int keyFileSz;
     FILE* keyFile = fopen(keyPath,"rb");
+    if (keyFile == NULL) {
+        printf("unable to open key file %s\n", keyPath);
+        return BAD_FUNC_ARG;
+    }
+
     fseek(keyFile, 0, SEEK_END);
     keyFileSz = ftell(keyFile);
     byte keyBuf[keyFileSz];
```

Comment:
I'm surprised this works. Wouldn't this buffer have to be dynamically allocated?